sdk/nodejs/appservice/zMixins.ts

```diff
@@ -415,6 +515,128 @@ export class CallbackFunctionApp<C extends Context<R>, E, R extends Result> exte
         this.container = container;
         this.plan = plan;
         this.zipBlob = zipBlob;
+
+        const routePrefix = args.hostSettings 
+            && args.hostSettings.extensions 
+            && args.hostSettings.extensions.http 
+            && args.hostSettings.extensions.http.routePrefix;
+        const rootPath = routePrefix === "" ? "" : `${routePrefix === undefined ? "api" : routePrefix}/`;
+
+        this.endpoint = pulumi.interpolate`https://${this.defaultHostname}/${rootPath}`;        
+    }
+}
+
+/**
+  * A CallbackFunctionApp is a special type of azure.appservice.FunctionApp that can be created out
+  * of an actual JavaScript function instance.  The function instance will be analyzed and packaged
+  * up (including dependencies) into a form that can be used by Azure Functions. See
+  * https://github.com/pulumi/docs/blob/master/reference/serializing-functions.md for additional
+  * details on this process.
+ */
+export class CallbackFunctionApp<C extends Context<R>, E, R extends Result> extends CombinedFunctionApp {
+    constructor(name: string, bindings: pulumi.Input<BindingDefinition[]>,
+                args: CallbackFunctionAppArgs<C, E, R>, opts: pulumi.CustomResourceOptions = {}) {
+        const functionsArgs = {
+            ...args,
+            functions: [<Function>{
+                name,
+                bindings,
+                callback: args,

```

Comment:
i'm a little confused that 'args' is both splatted with `...args` and passed to `callback`...  how does this work?

Reply:
Yeah, it's a side effect of extending both `CallbackArgs<C, E, R>` and `FunctionAppArgsBase`, so those are split here.

Reply:
👍